src/test/compile-fail/issue-51022.rs

```diff
@@ -0,0 +1,12 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// error-pattern: `main` function is not allowed to have lifetime parameters
+fn main<'a>() { }
```

Comment:
Instead of using `error-pattern` you can add below a line with
```
//~^ ERROR `main` function is not allowed to have lifetime parameters
```
instead.

Also, I'd prefer new tests to be `ui` instead of just `compile-fail`. That way we also test the visual output.

Reply:
Makes sense. Will do this is in a follow-up PR.

Reply:
Follow-up PR at https://github.com/rust-lang/rust/pull/51108